var/spack/repos/builtin/packages/wrf/package.py

```diff
@@ -0,0 +1,153 @@
+    depends_on('tcsh', type=('build'))
+    # time is not installed on all systems b/c bash provides it
+    # this fixes that for csh install scripts
+    depends_on('time', type=('build'))
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('m4', type='build')
+    depends_on('libtool', type='build')
+
+    def setup_build_environment(self, spack_env):
+        spack_env.set('NETCDF', self.spec['netcdf-c'].prefix)
+        if '+pnetcdf' in self.spec:
+            spack_env.set('PNETCDF', self.spec['parallel-netcdf'].prefix)
+        # This gets used via the applied patch files
+        spack_env.set('NETCDFF', self.spec['netcdf-fortran'].prefix)
+        spack_env.set('PHDF5', self.spec['hdf5'].prefix)
+        spack_env.set('JASPERINC', self.spec['jasper'].prefix.include)
+        spack_env.set('JASPERLIB', self.spec['jasper'].prefix.lib)
+
+        spack_env.prepend_path('PATH', dirname(self.compiler.fc))
+
+        if self.spec.satisfies('%gcc@10:'):
+            args = '-w -O2 -fallow-argument-mismatch -fallow-invalid-boz'
+            spack_env.set('FCFLAGS', args)
+            spack_env.set('FFLAGS', args)
+
+    def patch(self):
+        # Let's not assume csh is intalled in bin
+        files = glob.glob('*.csh')
+
+        filter_file('^#!/bin/csh -f', '#!/usr/bin/env csh', *files)
+        filter_file('^#!/bin/csh', '#!/usr/bin/env csh', *files)
+
+    def configure(self, spec, prefix):
+        build_opts = {"gcc":   {"serial": '32',
+                                "smpar":  '33',
+                                "dmpar":  '34',
+                                "dmsm":   '35'},
+                      "intel": {"serial": '13',
+                                "smpar":  '14',
+                                "dmpar":  '15',
+                                "dmsm":   '16'},
+                      "pgi":   {"serial": '52',
+                                "smpar":  '53',
+                                "dmpar":  '54',
+                                "dmsm":   '55'},
+                      }
+
+        nesting_opts = {"basic":  "1",
+                        "preset": "2",
+                        "vortex": "3"}
+
+        try:
+            compiler_opts = build_opts[self.spec.compiler.name]
+        except KeyError:
+            raise InstallError("Compiler not recognized nor supported.")
+
+        # Spack already makes sure that the variant value is part of the set.
+        build_type = compiler_opts[spec.variants['build_type'].value]
+
+        nesting_value = nesting_opts[spec.variants['nesting'].value]
+
+        with tempfile.TemporaryFile(mode='w') as fp:
+            fp.write(build_type + '\n' + nesting_value + '\n')
+            fp.seek(0)
+            Executable('./configure')(input=fp)
+
+    def build(self, spec, prefix):
+        csh = which('csh')
+        # num of processors for compile is capped at 20
+        csh('./compile', '-j', str(min(mp.cpu_count(), 20)),
```

Comment:
Spack provides a `make_jobs` argument that corresponds to the value of `spack install -j X`, you should use that instead.
```suggestion
        csh('./compile', '-j', str(min(int(make_jobs), 20)),
```

Reply:
Thanks @adamjstewart , this suggested change was added to latest commit.